runtime/near-vm-runner/src/runner.rs

```diff
@@ -1,16 +1,20 @@
-use near_primitives::checked_feature;
+use crate::vm_kind::VMKind;
 use near_primitives::config::VMConfig;
 use near_primitives::contract::ContractCode;
 use near_primitives::hash::CryptoHash;
-use near_primitives::profile::ProfileData;
 use near_primitives::runtime::fees::RuntimeFeesConfig;
 use near_primitives::types::CompiledContractCache;
 use near_primitives::version::ProtocolVersion;
-use near_vm_errors::VMError;
+use near_vm_errors::{FunctionCallError, VMRunnerError};
 use near_vm_logic::types::PromiseResult;
-use near_vm_logic::{External, ReturnData, VMContext, VMLogic, VMOutcome};
+use near_vm_logic::{External, VMContext, VMOutcome};
 
-use crate::vm_kind::VMKind;
+/// Returned by VM::run method.
+///
+/// `VMOutcome` is a graceful completion of a VM execution, which may or
+/// may not have failed. `VMRunnerError` contains fatal errors that usually
+/// are panicked on by the caller.
```

Comment:
This still sounds a bit confusing, is there perhaps a more useful wording? I think we want to underscore the differences between errors originating in the contract itself (contract divides by zero and dies) and in our runtime (runtime divides by zero and dies).